File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/dto/elementcache/ApiElementCache.java
package com.gigigo.orchextra.core.data.api.dto.elementcache;

import com.gigigo.orchextra.core.data.api.dto.elements.ApiElementSegmentation;
import java.util.List;

public class ApiElementCache {

  private String elementUrl;
  private String slug;
  private String type;
  private List<String> tags;
  private ApiElementCachePreview preview;
  private ApiElementCacheRender render;
  private ApiElementCacheShare share;
  private ApiElementSegmentation segmentation;
  private String name;
  private long updatedAt;

  public String getElementUrl() {
    return elementUrl;
  }

  public String getSlug() {
    return slug;
  }

  public String getType() {
    return type;
  }

  public List<String> getTags() {
    return tags;
  }

  public ApiElementCachePreview getPreview() {
    return preview;
  }

  public ApiElementCacheRender getRender() {
    return render;
  }

  public ApiElementCacheShare getShare() {
    return share;
  }

  public ApiElementSegmentation getSegmentation() {
    return segmentation;
  }

  public String getName() {
    return name;
  }

  public long getUpdatedAt() {
    return updatedAt;
  }

  public void setUpdatedAt(long updatedAt) {
    this.updatedAt = updatedAt;
  }
}

File: ocmsdk/src/main/java/com/gigigo/orchextra/core/data/api/dto/elementcache/ApiElementCache.kt
package com.gigigo.orchextra.core.data.api.dto.elementcache

class ApiElementCache(
    val slug: String,
    val type: String? = null,
    val tags: List<String>? = null,
    val preview: ApiElementCachePreview? = null,
    val render: ApiElementCacheRender? = null,
    val share: ApiElementCacheShare? = null,
    val customProperties: Map<String, Any>  = emptyMap(),
    val name: String? = null,
    var updatedAt: Long = 0) {
    companion object {
        fun empty() = ApiElementCache("")
    }
}
